src/HubManager.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry;
+
+use Sentry\State\Hub;
+use Sentry\State\HubInterface;
+
+final class HubManager implements HubManagerInterface
+{
+    /**
+     * @var HubInterface|null The current hub
+     */
+    private static $currentHub;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCurrentHub(): HubInterface
+    {
+        if (null === self::$currentHub) {
+            self::$currentHub = new Hub();
```

Comment:
Why returning a new Hub if it has not been defined yet?

One solution I see to avoid the pitfall of global state like this is to allow calling `setCurrentHub` only one time. Following calls would throw an exception.

`getCurrentHub` could then: throw an exception if no Hub is defined or return null. I would prefer throwing an exception but it's not a big deal.

Reply:
>Why returning a new Hub if it has not been defined yet?

We cannot throw an exception or return `null` because the behavior must be the same as calling `Hub::getCurrentHub()` with the current code. Also, creating the instance here means that the calling code does not need to handle creating the hub each time it needs it and this happens a lot of time (e.g. when calling the global functions). Finally, creating the instance outside this class means that the implementation would be hardcoded in places where the user cannot change it, instead here he can just (ideally) switch the hub manager implementation and do whatever he likes

>One solution I see to avoid the pitfall of global state like this is to allow calling `setCurrentHub` only one time

I don't have any valid reason about why you should not be able to call it more than once. Probably it makes more sense in other SDKs where the programming language needs more than one hub (e.g. one per thread) but the current API allows this and not permitting it anymore would be a BC :sob: